### PR TITLE
Setup drone

### PR DESCRIPTION
**Pull request by rjweeks**
This PR is for the Drone pipeline which is working and pushing images to Quay repository.

### CODE REVIEW

**Review by sathikmadetech — APPROVED** [review]
Looks good to me.

**Comment by LiamMacP** [comment]
This is missing a deploy step to get it into the environment.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
+kube/kd/cs-dev-migration-case-migrator-sqs.yaml
```
Comment (zuzak) [inline .gitignore:1]: What's this?

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+name: build
+trigger:
+  event:
+    - push
+  branch:
+    exclude:
+      - main
+
+steps:
+  - name: test project
```
Comment (LiamMacP) [inline .drone.yml:13]: You're missing the Localstack service here.

File: .drone.yml [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    image: quay.io/ukhomeofficedigital/hocs-base-image-build
+    environment:
+      SPRING_PROFILES_ACTIVE: "local"
+      LOCALSTACK_CONFIG_HOST: "localstack"
```
Comment (LiamMacP) [inline .drone.yml:17]: Unsure if this is required - maybe one to add if you do need it in the future for tests.

File: .drone.yml [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+      - cd config/localstack
+      - ./1-setup-sqs.sh
+      - ./2-setup-sns.sh
+    depends_on:
+      - clone
+
+  - name: test project
+    image: quay.io/ukhomeofficedigital/hocs-base-image-build
+    environment:
+      SPRING_PROFILES_ACTIVE: "local"
```
Comment (LiamMacP) [inline .drone.yml:74]: Same as above.

File: .drone.yml [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    environment:
+      VERSION: ${DRONE_COMMIT_SHA}
+      ENVIRONMENT: cs-dev-migration
+      KUBE_TOKEN:
+        from_secret: hocs_cms_data_migrator_cs_dev_migration
+      KUBE_SERVER: https://kube-api-notprod.notprod.acp.homeoffice.gov.uk  
+    commands:
+      - cd kube
+      - ./deploy.sh
+    depends_on:
```
Comment (LiamMacP) [inline .drone.yml:129]: You won't need this depends on as there is no other steps.

File: kube/kd/case-creator-queue-cm.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: hocs-case-creator-queue-config
+data:
+  # property-like keys; each key maps to a simple value
+  aws-account-id: "449472082214"
```
Comment (LiamMacP) [inline kube/kd/case-creator-queue-cm.yaml:7]: This shouldn't be on a public repo.

File: .drone.yml [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    image: quay.io/ukhomeofficedigital/localstack
+    environment:
+      SERVICES: sqs, sns
+      DEFAULT_REGION: eu-west-2
+
+steps:
+  - name: setup localstack
+    image: quay.io/ukhomeofficedigital/localstack
+    commands:
+      - cd config/localstack
```
Comment (SWhitlam) [inline .drone.yml:65]: you'll need to create the localstack folder and sqs and sns scripts here to create your queues

File: kube/kd/deployment.yaml [hunk 8]
```diff
@@ -152,59 +95,8 @@ spec:
...
-              cpu: 400m
-            requests:
-              memory: 96Mi
-              cpu: 100m
-
-
-        - name: hocs-casework
-          image: quay.digital.homeoffice.gov.uk/ukhomeofficedigital/hocs-casework:{{.VERSION}}
+        - name: hocs-cms-data-migrator
+          image: quay.digital.homeoffice.gov.uk/ukhomeofficedigital/hocs-cms-data-migrator:{{.VERSION}}
```
Comment (LiamMacP) [inline kube/kd/deployment.yaml:99]:
```suggestion
          image: quay.io/ukhomeofficedigital/hocs-cms-data-migrator:{{.VERSION}}
```

File: kube/kd/deployment.yaml [hunk 9]
```diff
@@ -222,90 +114,47 @@ spec:
...
-              value: 'https://hocs-audit.{{.KUBE_NAMESPACE}}.svc.cluster.local'
-            - name: HOCS_INFO_SERVICE
-              value: 'https://hocs-info-service.{{.KUBE_NAMESPACE}}.svc.cluster.local'
-            - name: HOCS_SEARCH_SERVICE
-              value: 'https://hocs-search.{{.KUBE_NAMESPACE}}.svc.cluster.local'
-            - name: HOCS_DOCUMENT_SERVICE
-              value: 'https://hocs-docs.{{.KUBE_NAMESPACE}}.svc.cluster.local'
-            - name: HOCS_BASICAUTH
+              value: 'sqs, s3'
+            - name: AWS_SQS_CASE_CREATOR_ACCOUNT_ACCESS_KEY
```
Comment (LiamMacP) [inline kube/kd/deployment.yaml:118]: CASE_CREATOR?

File: kube/kd/service.yaml [hunk 10]
```diff
@@ -0,0 +1,16 @@
+
```
Comment (LiamMacP) [inline kube/kd/service.yaml:1]: Nitpick: L1-2 don't need to be present.

File: .gitignore (hunk 1 above)
Comment (LiamMacP) [inline .gitignore:1]: Added to .gitignore - but you have added a file `case-creator-queue-cm.yaml`? This doesn't feel right?

File: kube/kd/autoscale.yaml [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+  labels:
+    app: hocs-cms-data-migrator
+  name: hocs-cms-data-migrator
+spec:
+  maxReplicas: {{.MAX_REPLICAS}}
+  minReplicas: {{.MIN_REPLICAS}}
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: hocs-case-creator
```
Comment (LiamMacP) [inline kube/kd/autoscale.yaml:13]: Wrong name here...

File: .drone.yml (hunk 2 above)
Comment (LiamMacP) [reply .drone.yml:13]: Also missing the setup step.

**Comment by LiamMacP** [comment]
You'll want a README for this too.

**Comment by LiamMacP** [comment]
Can you add the .drone.yml for K8 config linting too please.

**Comment by LiamMacP** [comment]
You're utilising tags in Drone. You'll want the GitHub actions present for this. Also want code-ql and Dependabot GitHub actions too.

File: kube/kd/case-creator-queue-cm.yaml (hunk 6 above)
Comment (rjweeks) [reply kube/kd/case-creator-queue-cm.yaml]: Removed.

File: .gitignore (hunk 1 above)
Comment (rjweeks) [reply .gitignore]: Removed from Repo.

File: .drone.yml (hunk 5 above)
Comment (rjweeks) [reply .drone.yml]: Removed.

File: .drone.yml (hunk 2 above)
Comment (rjweeks) [reply .drone.yml:13]: Removed localstack from env for now as there is a separate ticket for Localstack.

File: .drone.yml [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+    - push
+  branch:
+    exclude:
+      - main
+
+steps:
+  - name: test project
+    image: quay.io/ukhomeofficedigital/hocs-base-image-build
+    environment:
+      SPRING_PROFILES_ACTIVE: "local"
```
[the remark this answers is not in the document]
Comment (rjweeks) [reply .drone.yml:16]: Removed.

File: .drone.yml (hunk 3 above)
Comment (rjweeks) [reply .drone.yml:17]: Removed.

File: .drone.yml (hunk 4 above)
Comment (rjweeks) [reply .drone.yml:74]: Removed.

File: .drone.yml (hunk 7 above)
Comment (rjweeks) [reply .drone.yml:65]: Removed localstack for now.

File: kube/kd/deployment.yaml (hunk 9 above)
Comment (rjweeks) [reply kube/kd/deployment.yaml:118]: Changed to CMS_DATA_MIGRATOR

File: kube/kd/service.yaml (hunk 10 above)
Comment (rjweeks) [reply kube/kd/service.yaml:1]: Removed.

File: src/main/resources/application.properties [hunk 13]
```diff
@@ -10,4 +10,6 @@ spring.jpa.properties.hibernate.format_sql = true
 spring.jpa.properties.hibernate.dialect = org.hibernate.dialect.SQLServer2016Dialect
 
 # Hibernate ddl auto (create, create-drop, validate, update)
-spring.jpa.hibernate.ddl-auto = update
\ No newline at end of file
+spring.jpa.hibernate.ddl-auto = update
+
+management.health.probes.enabled=true
```
[the remark this answers is not in the document]
Comment (rjweeks) [reply src/main/resources/application.properties:15]: Done.

File: kube/kd/deployment.yaml [hunk 14]
```diff
@@ -222,90 +114,47 @@ spec:
...
-            - name: HOCS_AUDIT_SERVICE
-              value: 'https://hocs-audit.{{.KUBE_NAMESPACE}}.svc.cluster.local'
-            - name: HOCS_INFO_SERVICE
-              value: 'https://hocs-info-service.{{.KUBE_NAMESPACE}}.svc.cluster.local'
-            - name: HOCS_SEARCH_SERVICE
-              value: 'https://hocs-search.{{.KUBE_NAMESPACE}}.svc.cluster.local'
-            - name: HOCS_DOCUMENT_SERVICE
-              value: 'https://hocs-docs.{{.KUBE_NAMESPACE}}.svc.cluster.local'
-            - name: HOCS_BASICAUTH
+              value: 'sqs, s3'
```
[the remark this answers is not in the document]
Comment (rjweeks) [reply kube/kd/deployment.yaml:117]: Removed.

File: kube/kd/deployment.yaml [hunk 15]
```diff
@@ -222,90 +114,47 @@ spec:
...
-            - name: AWS_SQS_NOTIFY_ACCOUNT_SECRET_KEY
-              valueFrom:
-                secretKeyRef:
-                  name: {{.KUBE_NAMESPACE}}-notify-sqs
-                  key: secret_access_key
+                  name: {{.KUBE_NAMESPACE}}-mssql
+                  key: password
           resources:
             limits:
               cpu: 2000m
```
[the remark this answers is not in the document]
Comment (rjweeks) [reply kube/kd/deployment.yaml:158]: Reduced.

File: README.md [hunk 16]
```diff
@@ -0,0 +1 @@
+# hocs-cms-migrator
```
Comment (LiamMacP) [inline README.md:1]: Feels like there should be more in this, licensing etc.

File: kube/deploy.sh [hunk 17]
```diff
@@ -3,11 +3,10 @@ set -euo pipefail
 
 export KUBE_NAMESPACE=${ENVIRONMENT}
 export KUBE_SERVER=${KUBE_SERVER}
-export KUBE_TOKEN=${KUBE_TOKEN}
```
Comment (LiamMacP) [inline kube/deploy.sh:6]: This can be left it as a earlier exit with the `set [options]`.

File: kube/kd/deployment.yaml [hunk 18]
```diff
@@ -213,102 +105,57 @@ spec:
                 - SETGID
           envFrom:
             - configMapRef:
-                name: hocs-queue-config
+                name: hocs-case-creator-queue-config
           env:
             - name: JAVA_OPTS
               value: '-Xms1280m -Xmx1280m -Djavax.net.ssl.trustStore=/etc/keystore/truststore.jks -Dhttps.proxyHost=hocs-outbound-proxy.{{.KUBE_NAMESPACE}}.svc.cluster.local -Dhttps.proxyPort=31290 -Dhttp.nonProxyHosts=*.{{.KUBE_NAMESPACE}}.svc.cluster.local'
```
Comment (LiamMacP) [inline kube/kd/deployment.yaml:111]: Are the `Xms` and `Xmx` set to sensible values?

Comment (LiamMacP) [reply kube/kd/deployment.yaml:111]: Do you have an outbound proxy on deployment? I haven't checked is all.

File: kube/kd/deployment.yaml [hunk 19]
```diff
@@ -213,102 +105,57 @@ spec:
...
-                secretKeyRef:
-                  name: {{.KUBE_NAMESPACE}}-notify-sqs
-                  key: secret_access_key
+                  name: {{.KUBE_NAMESPACE}}-mssql
+                  key: password
           resources:
             limits:
-              cpu: 2000m
+              cpu: 500m
               memory: 2204Mi
```
Comment (LiamMacP) [inline kube/kd/deployment.yaml:159]: Same for memory, is this high?

**Review by LiamMacP — COMMENTED** [review]
Few minor things to look at but happy to approve after you've checked.

File: build.gradle [hunk 20]
```diff
@@ -13,7 +13,9 @@ repositories {
 }
```
Comment (robertdeniszczyc2) [inline build.gradle:14]: Think this should be using spring boot 2.7.0 and java 17 rather than 2.6.7 and 11?

File: README.md (hunk 16 above)
Comment (robertdeniszczyc2) [inline README.md:1]: Everywhere else the module is referenced as hocs-cms-data-migrator not hocs-cms-migrator

File: kube/kd/deployment.yaml [hunk 21]
```diff
@@ -130,7 +73,7 @@ spec:
                 - SETGID
           args:
             - --certs=/certs
-            - --domain=hocs-casework.${KUBE_NAMESPACE}.svc.cluster.local
+            - --domain=hocs-cms-migrator.${KUBE_NAMESPACE}.svc.cluster.local
```
Comment (robertdeniszczyc2) [inline kube/kd/deployment.yaml:76]: This should be hocs-cms-data-migrator

File: src/main/resources/application.properties (hunk 13 above)
Comment (robertdeniszczyc2) [inline src/main/resources/application.properties:15]:
```suggestion
management.endpoint.health.probes.enabled=true
```

File: src/main/resources/application.properties [hunk 22]
```diff
@@ -10,4 +10,6 @@ spring.jpa.properties.hibernate.format_sql = true
 spring.jpa.properties.hibernate.dialect = org.hibernate.dialect.SQLServer2016Dialect
 
 # Hibernate ddl auto (create, create-drop, validate, update)
-spring.jpa.hibernate.ddl-auto = update
\ No newline at end of file
+spring.jpa.hibernate.ddl-auto = update
```
Comment (robertdeniszczyc2) [inline src/main/resources/application.properties:13]: Is using `update` here desirable? Which schema is this acting against as we don't want to update the Lagan schema.

File: README.md (hunk 16 above)
Comment (rjweeks) [reply README.md:1]: I've added license details and a bit of an introduction. The readme will be updated as development work is completed.

Comment (rjweeks) [reply README.md]: Updated.

File: kube/kd/deployment.yaml (hunk 21 above)
Comment (rjweeks) [reply kube/kd/deployment.yaml]: Updated.

File: kube/deploy.sh (hunk 17 above)
Comment (rjweeks) [reply kube/deploy.sh:6]: Left this in now.

File: kube/kd/deployment.yaml (hunk 19 above)
Comment (rjweeks) [reply kube/kd/deployment.yaml:159]: Reduced.

File: build.gradle (hunk 20 above)
Comment (LiamMacP) [reply build.gradle:14]: We could have it changed here, but really we should have Dependabot added and let that handle it for us.

Comment (rjweeks) [reply build.gradle:14]: Updated.

File: src/main/resources/application.properties (hunk 22 above)
Comment (rjweeks) [reply src/main/resources/application.properties:13]: I've changed to none as we don't want to update the Lagan schema.

File: src/main/resources/application.properties (hunk 13 above)
Comment (rjweeks) [reply src/main/resources/application.properties]: Updated.

**Comment by robertdeniszczyc2** [comment]
Changes I commented on look to be fixed

**Review by LiamMacP — APPROVED** [review]

**Pull request merged**